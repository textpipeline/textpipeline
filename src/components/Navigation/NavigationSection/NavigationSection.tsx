import Box from '@material-ui/core/Box';
import Divider from '@material-ui/core/Divider';
import React from 'react';

export interface NavigationSectionProps {
  readonly isPrimary?: boolean;
}

const NavigationSection: React.FC<NavigationSectionProps> = ({ children, isPrimary }) => (
  <Box flexGrow={isPrimary ? 1 : 0}>
    <Divider />
    {children}
  </Box>
);

export default NavigationSection;
