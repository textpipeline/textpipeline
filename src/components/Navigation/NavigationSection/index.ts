export { default } from './NavigationSection';
export * from './NavigationSection';
