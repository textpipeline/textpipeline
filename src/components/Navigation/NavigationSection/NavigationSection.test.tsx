import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import NavigationSection from './NavigationSection';

describe('NavigationSection', () => {
  let rendered: ShallowWrapper;

  describe('when isPrimary is not specified', () => {
    beforeEach(() => {
      rendered = shallow(<NavigationSection />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('when isPrimary is specified false', () => {
    beforeEach(() => {
      rendered = shallow(<NavigationSection isPrimary={false} />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('when isPrimary is specified true', () => {
    beforeEach(() => {
      rendered = shallow(<NavigationSection isPrimary={true} />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });
});
