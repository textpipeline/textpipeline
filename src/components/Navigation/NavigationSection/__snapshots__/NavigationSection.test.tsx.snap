// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationSection when isPrimary is not specified renders correctly 1`] = `
<Styled(MuiBox)
  flexGrow={0}
>
  <WithStyles(ForwardRef(Divider)) />
</Styled(MuiBox)>
`;

exports[`NavigationSection when isPrimary is specified false renders correctly 1`] = `
<Styled(MuiBox)
  flexGrow={0}
>
  <WithStyles(ForwardRef(Divider)) />
</Styled(MuiBox)>
`;

exports[`NavigationSection when isPrimary is specified true renders correctly 1`] = `
<Styled(MuiBox)
  flexGrow={1}
>
  <WithStyles(ForwardRef(Divider)) />
</Styled(MuiBox)>
`;
